File: src/main/java/com/iminling/model/exception/AuthorizeException.java
package com.iminling.model.exception;

import com.iminling.model.common.MessageCode;
import lombok.Getter;
import lombok.Setter;

/**
 * 用户认证异常类
 * @author yslao@outlook.com
 */
public class AuthorizeException extends RuntimeException {

    @Getter @Setter private MessageCode messageCode;

    public AuthorizeException(){
        super(MessageCode.PERMISSION_DENY.getMessage());
        this.messageCode = MessageCode.PERMISSION_DENY;
    }

    public AuthorizeException(MessageCode messageCode){
        super(messageCode.getMessage());
        this.messageCode = messageCode;
    }

}

File: iminling-core/src/main/kotlin/com/iminling/core/exception/AuthorizeException.kt
package com.iminling.core.exception

import com.iminling.core.constant.MessageCode

/**
 * 用户认证异常类
 * @author  yslao@outlook.com
 * @since  2021/11/26
 */
class AuthorizeException(open val messageCode: MessageCode): RuntimeException(messageCode.message) {

    constructor(): this(MessageCode.PERMISSION_DENY)

}